fabtests/common/shared.c

```diff
@@ -1240,8 +1285,19 @@ int ft_exchange_keys(struct fi_rma_iov *peer_iov)
 		} else {
 			rma_iov->addr = 0;
 		}
-		rma_iov->key = fi_mr_key(mr);
-		ret = ft_tx(ep, remote_fi_addr, sizeof *rma_iov, &tx_ctx);
+
+		if (fi->domain_attr->mr_mode & FI_MR_RAW) {
+			/* Get raw attributes */
+			ret = fi_mr_raw_attr(mr, &addr,
+					(uint8_t *) &rma_iov->key, &key_size,
+					0);
+			if (ret)
+				return ret;
+		} else {
+			rma_iov->key = fi_mr_key(mr);
+		}
+
+		ret = ft_tx(ep, remote_fi_addr, len, &tx_ctx);
```

Comment:
This function is getting too long.  Can you create a new call, ft_exchange_raw_keys(), and move all raw key handling there?  Then we can have a single check for FI_MR_RAW to determine which function to call.